test/instrumentation/index.js

```diff
@@ -653,6 +653,27 @@ test('nested transactions', function (t) {
   t0.end()
 })
 
+test('does not record requests from internals', function (t) {
+  resetAgent(3, function (data) {
```

Comment:
Instead of waiting for exactly 3 events, this will make the mock agent wait about 100ms after the last received event and then call the callback if there have been no other events:

```suggestion
  resetAgent(function (data) {
```

Then we should just check that `data` contains the 1 metadata object, 2 spans, and 1 transaction that we expected. I think that should work.

Reply:
fixed